### PR TITLE
Script for fetching cursors kinds from Index.h

### PR DESCRIPTION
**Pull request by graywolf**
This script was created in reaction to the original comment:

    # Manually extracted from Index.h
    # Doing it by hand is long, error prone and horrible, we must find a way
    # to do that automatically.

It seems to do the job. Just pass the path of `Index.h` as first argument and it regenerates the new `kinds.py` file holding the dictionary.

Could I possibly get some feedback with the end-goal of getting this merged in?

### CODE REVIEW

File: bin/generate_kinds.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+kinds = None
+for child in tu.cursor.get_children():
+  if (child.spelling == "CXCursorKind"):
+    kinds = child
+    break
+else:
+  print "Index.h doesn't contain CXCursorKind where it is expected, please report a bug."
+  exit(-1)
+
+kinds_py_path = os.path.join(
```
Comment (xaizek) [inline bin/generate_kinds.py]: `os.path.join()` accepts more than 2 arguments, so one call can be removed.

File: bin/generate_kinds.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  f.write("# !! GENERATED FILE, DO NOT EDIT\n")
+  f.write("kinds = {\n")
+
+  for kind in kinds.get_children():
+    # filter out First/Last markers from the enum
+    if fl.match(kind.spelling) is not None:
+      continue
+
+    if kind.enum_value in mapping:
```
Comment (xaizek) [inline bin/generate_kinds.py]: There is a handy [get method](https://docs.python.org/2/library/stdtypes.html#dict.get) which would shorten this statement to:
```python
text = mapping.get(kind.enum_value, kind.enum_value)
```

**Comment by xaizek** [comment]
Looks good to me. Tested it and don't have anything to add except those two nitpicks. As everyone else with commit access seem to ignore this repository, I think there won't be any objections to merging this in :-)

File: bin/generate_kinds.py (hunk 1 above)
Comment (graywolf) [inline bin/generate_kinds.py]: I was pretty sure there was a better way to do it, so thanks for teaching me :)

**Comment by graywolf** [comment]
ok, I implemented both your suggestions, so hopefully it looks all good now ^_^

**Comment by xaizek** [comment]
Thanks!

**Pull request merged**